components/prism-n1ql.js

```diff
@@ -0,0 +1,22 @@
+Prism.languages.n1ql = {
+	'comment': /(?:\/\*[\s\S]*?\*\/)/,
+	'parameter': [
+		/\$[\w.]+/
+	],
+	'string': {
+		pattern: /(^|[^@\\])("|')(?:\\[\s\S]|(?!\2)[^\\]|\2\2)*\2/,
```

Comment:
`string` and `identifier` both cannot be preceded by an `@`. I'm guessing that you copied this from SQL where strings and variables interfere with each other but since this is not the case in N1QL, you can remove it.

The backslash is also bugging me. I don't understand why it is there neither here nor in SQL. Am I missing something?

Reply:
You're completely right with the `@`. I did borrow heavily from the SQL regexes here, which is why this slipped in. 
I believe the backslash is so you cannot have a single backslash at the end of the string. 
For example, this prevents us matching `"foo\"`, which is an invalid string due to the last quotation mark.

Reply:
If I were to guess it's probably a relic of the time when Prism did not support greedy matching.
We generally don't worry about highlighting syntactically incorrect code, so that should not concern us.
It's probably safe to remove the whole lookbehind.